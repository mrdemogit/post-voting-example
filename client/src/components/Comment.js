import React, { memo } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import VoteSection from './VoteSection'
import { CommentType } from '../types'

const StyledComment = styled.div`
  padding: 0.5em 1em;
  margin: 0.2em;
  background: #04597ad4;
  border-radius: 0.3em;
  color: #fff;
`

const StyledName = styled.div`
  color: #000;
  font-style: italic;
`

export const Comment = ({ uuid, author, content, votes, upvote, downvote }) => (
  <StyledComment>
    <StyledName>{author}</StyledName>
    <div>{content}</div>
    <VoteSection
      uuid={uuid}
      votes={votes}
      upvote={upvote}
      downvote={downvote}
    />
  </StyledComment>
)

Comment.propTypes = {
  ...CommentType,
  upvote: PropTypes.func.isRequired,
  downvote: PropTypes.func.isRequired
}

export default memo(Comment)
