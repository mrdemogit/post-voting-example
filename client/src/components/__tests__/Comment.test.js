import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { Comment } from '../Comment'

describe('Comment', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer()
    renderer.render(
      <Comment
        uuid='uuid1'
        author='authorTest'
        content='contentTest'
        votes={123}
        upvote={jest.fn()}
        downvote={jest.fn()}
      />
    )

    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
