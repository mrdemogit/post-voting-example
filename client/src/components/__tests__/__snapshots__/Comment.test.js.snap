// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment should render correctly 1`] = `
<ForwardRef>
  <ForwardRef>
    authorTest
  </ForwardRef>
  <div>
    contentTest
  </div>
  <VoteSection
    downvote={[MockFunction]}
    upvote={[MockFunction]}
    uuid="uuid1"
    votes={123}
  />
</ForwardRef>
`;
